src/libstd/fs.rs

```diff
@@ -1069,13 +1069,23 @@ pub fn remove_dir_all<P: AsRef<Path>>(path: P) -> io::Result<()> {
 }
 
 fn _remove_dir_all(path: &Path) -> io::Result<()> {
+    let filetype = try!(symlink_metadata(path)).file_type();
+    if filetype.is_symlink() {
+        remove_file(path)
```

Comment:
Well, the code that removed a symlink inside the directory used `remove_file` too, AFAICS. So at least it's not a regression.

Anyway, it would be helpful if someone can test it on windows.